fe/fe-core/src/main/java/com/starrocks/scheduler/TaskManager.java

```diff
@@ -173,14 +241,204 @@ public ShowResultSet handleSubmitTaskStmt(SubmitTaskStmt submitTaskStmt) throws
             }
             throw new DdlException("Failed to create Task: " +  taskName + ", ErrorCode: " + createResult);
         }
-        String queryId = this.executeTask(taskName);
+        SubmitResult submitResult = executeTask(taskName);
+        if (submitResult.getStatus() != SubmitResult.SubmitStatus.SUBMITTED) {
+            dropTasks(ImmutableList.of(task.getId()), false);
+        }
 
         ShowResultSetMetaData.Builder builder = ShowResultSetMetaData.builder();
         builder.addColumn(new Column("TaskName", ScalarType.createVarchar(40)));
         builder.addColumn(new Column("Status", ScalarType.createVarchar(10)));
-        List<String> item = ImmutableList.of(taskName, "Submitted");
+        List<String> item = ImmutableList.of(taskName, submitResult.getStatus().toString());
         List<List<String>> result = ImmutableList.of(item);
         return new ShowResultSet(builder.build(), result);
     }
 
+    public long loadTasks(DataInputStream dis, long checksum) throws IOException {
+        int taskCount = 0;
+        try {
+            String s = Text.readString(dis);
+            SerializeData data = GsonUtils.GSON.fromJson(s, SerializeData.class);
+            if (data != null) {
+                if (data.tasks != null) {
+                    for (Task task : data.tasks) {
+                        replayCreateTask(task);
+                    }
+                    taskCount = data.tasks.size();
+                }
+
+                if (data.runStatus != null) {
+                    for (TaskRunStatus runStatus : data.runStatus) {
+                        replayCreateTaskRun(runStatus);
+                    }
+                }
+            }
+            LOG.info("finished replaying TaskManager from image");
+        } catch (EOFException e) {
+            LOG.info("no TaskManager to replay.");
```

Comment:
If there is no TaskManager's image. You should not update the checksum, or will fail to check the checksum on the end of load image.